src/workspace-tree/icons.ts

```diff
@@ -60,23 +59,19 @@ const SPECIFIC_RULE_CLASS_ICONS: Record<string, string> = {
  */
 export function getBazelRuleIcon(
   target: blaze_query.ITarget,
-): string | vscode.ThemeIcon {
+): IconName | undefined {
```

Comment:
just a word of caution: `strictNullChecks` is currently disabled in our `tsconfig.json`. So the Typescript compiler doesn't actually check those type annotations closely.

This is still legacy, we didn't find time to clean up the code such that we could enable `strictNullChecks` , yet

(I am not asking you to implement this as part of this commit - just as food for thought 🙂)

Reply:
Do you mean this is not `undefined`? I thought when `iconName` is not found in `SPECIFIC_RULE_CLASS` and either matches in the following suffix rules, then it would be undefined.

Reply:
your annotations are correct. But the TypeScript compiler does not verify them, yet, because we opted out of that check

Reply:
Noted. Let me try enabled it locally while developing new changes.